xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Attachments.xml

```diff
@@ -0,0 +1,1144 @@
+      <defaultCategory>confluence</defaultCategory>
+    </property>
+    <property>
+      <description/>
+    </property>
+    <property>
+      <id>attachments</id>
+    </property>
+    <property>
+      <name>attachments</name>
+    </property>
+    <property>
+      <priority/>
+    </property>
+    <property>
+      <supportsInlineMode>0</supportsInlineMode>
+    </property>
+    <property>
+      <visibility>Current Wiki</visibility>
+    </property>
+  </object>
+  <object>
+    <name>Confluence.Macros.Attachments</name>
+    <number>0</number>
+    <className>XWiki.WikiMacroParameterClass</className>
+    <guid>d584bdf4-8906-4c1c-bcf5-1a89a66021d4</guid>
+    <class>
+      <name>XWiki.WikiMacroParameterClass</name>
+      <customClass/>
+      <customMapping/>
+      <defaultViewSheet/>
+      <defaultEditSheet/>
+      <defaultWeb/>
+      <nameField/>
+      <validationScript/>
+      <defaultValue>
+        <disabled>0</disabled>
+        <name>defaultValue</name>
+        <number>4</number>
+        <prettyName>Parameter default value</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </defaultValue>
+      <description>
+        <disabled>0</disabled>
+        <name>description</name>
+        <number>2</number>
+        <prettyName>Parameter description</prettyName>
+        <rows>5</rows>
+        <size>40</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </description>
+      <mandatory>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>mandatory</name>
+        <number>3</number>
+        <prettyName>Parameter mandatory</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </mandatory>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>1</number>
+        <prettyName>Parameter name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <type>
+        <disabled>0</disabled>
+        <name>type</name>
+        <number>5</number>
+        <prettyName>Parameter type</prettyName>
+        <size>60</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </type>
+    </class>
+    <property>
+      <defaultValue/>
+    </property>
+    <property>
+      <description>A comma-separated list of regular expressions, used to filter the attachments by file name. Note that the parameter values must be regular expressions. For example:
+
+* To match a file suffix of 'jpg', use .*jpg (not *.jpg).
+* To match file names ending in 'jpg' or 'png', use .*jpg,.*png
```

Comment:
Are you sure this looks good in the WYSIWYG editor Edit Macro modal? I'm asking because wiki syntax is not rendered there AFAIK. The macro parameter description is displayed as plain text. The line endings should also be lost because white spaces are collapsed in HTML (and ``\n`` doesn't break the line unless white spaces are preserved, but it's not the case here).

Reply:
Indeed, I figured this out too and I forgot to point it out!
So actually why are macro parameter description fields using the wysiwyg editor if they are rendered as plain text? From what you said I can expect this is not a bug but the intended behavior. But in that case this is quite confusing to have a wysiwyg editor here, especially considering that the macro description field is using a plain text editor on its side.

Reply:
I'll try to update macro parameter descriptions to plain text, but this is going to be difficult for some of them which have rather complex/long descriptions like `sortBy`

Reply:
> why are macro parameter description fields using the wysiwyg editor

That's a bug IMO. See https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-wikimacro/xwiki-platform-rendering-wikimacro-store/src/main/java/org/xwiki/rendering/wikimacro/internal/WikiMacroClassDocumentInitializer.java#L77 versus https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-wikimacro/xwiki-platform-rendering-wikimacro-store/src/main/java/org/xwiki/rendering/wikimacro/internal/WikiMacroParameterClassDocumentInitializer.java#L55 . We should force the PURE_TEXT editor for parameter description field also.